docs/composition.md

```diff
@@ -0,0 +1,882 @@
+---
+title: Publishing Infrastructure
+toc: true
+weight: 400
+indent: true
+---
+
+# Publishing Infrastructure
+
+Crossplane allows infrastructure operators to define new kinds of infrastructure
+resources and publish them for the application operators they support to use,
+all without writing any code.
+
+Infrastructure providers enable Crossplane to manage a wide array of
+infrastructure resources like Azure SQL servers and AWS ElastiCache clusters.
+Infrastructure composition allows infrastructure operators to define, share, and
+reuse new kinds of infrastructure resources that are _composed_ of other
+infrastructure resources. Infrastructure operators may configure one or more
+compositions of any defined resource, and may publish any defined resource to
+their application operators, who may then declare that their application
+requires that kind of resource.
+
+Composition can be used to build a catalogue of kinds and configuration classes
+of infrastructure that fit the needs and opinions of your organisation. As an
+infrastructure operator you might define your own `MySQLInstance` resource. This
+resource would allow your application operators to configure only the settings
+that _your_ organisation needs - perhaps engine version and storage size. All
+other settings are deferred to a selectable composition representing a
+configuration class like "production" or "staging".
+
+> Note that composition is an **experimental** feature of Crossplane. Refer to
+> [Current Limitations] for information on functionality that is planned but not
+> yet implemented.
+
+## Concepts
+
+![Infrastructure Composition Concepts]
+
+A _Composite_ infrastructure resource is composed of other resources. Its
+configuration schema is user-defined. The `MySQLInstance` resources in the above
+diagram are composite infrastructure resources.
+
+A `Composition` specifies a set of resources of which a composite infrastructure
+resource may be composed. The Azure and AWS `Composition` resources in the above
```

Comment:
The `Composition` CRD connects the "composite infrastructure resource", eg `MySQLInstance`, to the provider-specific set of managed resources needed to fulfill the promise of that resource, eg in the case of AWS, `DBSubnetGroup` + `RDSInstance`. The Composition also specifies how to copy values needed by these managed resources from the "composite infrastructure resource", using the `fromFieldPath` -> `toFieldPath` keywords.

Reply:
What about "compositions configure composite composition?" :trollface:

Reply:
I like the idea of including an example of how a `MySQLInstance` gets reconciled with a particular set of composed resources - will push an update.